backend/app/routes/web/search_results.py

```diff
@@ -87,9 +88,13 @@ async def search_post(
         if collection in [Collection.ALL_COLLECTION, Collection.BUNDLE]:
             await extend_results_with_bundles(client, res_json)
         if collection in [Collection.ALL_COLLECTION, Collection.GUIDELINE]:
-            res_json["response"]["docs"] = await extend_ig_with_related_services(
-                client, res_json["response"]["docs"]
-            )
+            try:
+                new_docs = await extend_ig_with_related_services(
+                    client, res_json["response"]["docs"]
+                )
+                res_json["response"]["docs"] = copy.deepcopy(new_docs)
```

Comment:
Is it necessary to create everywhere a deepcopy? We should definitely avoid that as much as possible. That is a costly operation